SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_multiple_classes_WHEN_generated_schema_for_these_classes_THEN_corresponding_schema_files_exist)
+{
+	// GIVEN
+	TSet<UClass*> Classes =
+	{
+		USpatialTypeObjectStub::StaticClass(),
+		ASpatialTypeActor::StaticClass()
+	};
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	bool bExpectedFilesExist = true;
+	for (const auto& CurrentClass : Classes)
+	{
+		if (LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass).IsEmpty())
+		{
+			bExpectedFilesExist = false;
+			break;
+		}
+	}
+
+	TestTrue("All expected schema files have been generated", bExpectedFilesExist);
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_an_Actor_class_WHEN_generated_schema_for_this_class_THEN_a_file_with_valid_schema_exists)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	UClass* CurrentClass = ASpatialTypeActor::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+	TestTrue("Generated Actor schema is valid", Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass));
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_multiple_Actor_classes_WHEN_generated_schema_for_these_classes_THEN_files_with_valid_schema_exist)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	TSet<UClass*> Classes =
+	{
+		ASpatialTypeActor::StaticClass(),
+		ANonSpatialTypeActor::StaticClass()
+	};
+
+	// Classes need to be sorted to have proper ids
+	Classes.Sort([](const UClass& A, const UClass& B)
+	{
+		return A.GetPathName() < B.GetPathName();
+	});
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	bool bValidSchemaExists = true;
+	for (const auto& CurrentClass : Classes)
+	{
+		FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+		if(!Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass))
+		{
+			bValidSchemaExists = false;
+			break;
+		}
+	}
+
+	TestTrue("Generated Actor schema is valid", bValidSchemaExists);
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_an_Actor_component_class_WHEN_generated_schema_for_this_class_THEN_a_file_with_valid_schema_exists)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	UClass* CurrentClass = USpatialTypeActorComponent::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
```

Comment:
Is this something the schema generator does explicitly? If so we should make it into a function and reuse it here probably.

Reply:
Discussed, we don't want to have a sort function in SchemaGen interface.